aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -867,6 +867,69 @@ std::tuple<Tensor, Tensor> _nested_compute_contiguous_strides_offsets(const Tens
       construct_offsets(nested_size));
 }
 
+Tensor _nested_strided_to_jagged(const Tensor& self) {
+  auto self_ptr = get_nested_tensor_impl(self);
+
+  // All jagged NT can be converted into strided NTs, but the opposite is not True
+  // Only strided NTs with a single jagged dimension might be converted into
+  // jagged NTs, so first we check for that
+  int ragged_dims_count = 0;
+  int ragged_idx = -1;
+  for (int64_t i = 0; i < self_ptr->dim(); ++i) {
+    if (!self_ptr->opt_size(i).has_value()) {
+      ragged_dims_count++;
+      ragged_idx = i;
+    }
+  }
+  TORCH_INTERNAL_ASSERT(ragged_dims_count == 1, "Only strided NTs with 1 ragged dim can be converted to jagged NTs");
+
+  // Once that's checked, we convert the offsets + sizes in strided NT to
+  // offsets + (optionally) lengths for the jagged NT
+  auto ragged_offsets = self_ptr->get_storage_offsets();
+  const int64_t* ragged_offsets_ptr = ragged_offsets.const_data_ptr<int64_t>();
+  auto ragged_sizes = self_ptr->get_nested_sizes();
+  const int64_t* ragged_sizes_ptr = ragged_sizes.const_data_ptr<int64_t>();
+  int64_t post_ragged_stride = 1;
+  for (int64_t i : c10::irange(ragged_idx, ragged_sizes.size(1))) {
+    post_ragged_stride *= ragged_sizes_ptr[i];
+  }
+  auto ragged_offsets_sizes = ragged_offsets.sizes();
+  auto metadata_tensor_options = self_ptr->get_buffer().options().dtype(kInt);
+  auto jagged_offsets = at::empty({ragged_offsets_sizes[0]+1}, metadata_tensor_options);
```

Comment:
I think in general we expect jagged offsets to be int64 but I'm seeing int32 and the conversion